code/modules/client/preference_setup/controls/01_keybindings.dm

```diff
@@ -0,0 +1,254 @@
+				var/list/default_keys = pref.hotkeys ? kb.hotkey_keys : kb.classic_keys
+				. += {"<td width='15%'><a [user_binds[kb.name] ~= default_keys ? "class='linkOff fluid'" : "class='fluid' href ='?src=\ref[src];preference=keybinding_reset;keybinding=[kb.name];old_keys=[jointext(user_binds[kb.name], ",")]"]'>Reset</a></td>"}
+				. += "</tr>"
+		. += "</table>"
+
+	. += "</div>"
+	. += "<br><br>"
+	. += "<a href ='?src=\ref[src];preference=keybindings_reset'>Reset to default</a>"
+	. += "</center>"
+
+	return jointext(., null)
+
+/datum/category_item/player_setup_item/controls/keybindings/proc/capture_keybinding(mob/user, datum/keybinding/kb, old_key)
+	var/HTML = {"
+	<div class='Section fill'id='focus' style="outline: 0; text-align:center;" tabindex=0>
+		Keybinding: [kb.full_name]<br>[kb.description]
+		<br><br>
+		<b>Press any key to change<br>Press ESC to clear</b>
+	</div>
+	<script>
+	var deedDone = false;
+	document.onkeyup = function(e) {
+		if(deedDone){ return; }
+		var alt = e.altKey ? 1 : 0;
+		var ctrl = e.ctrlKey ? 1 : 0;
+		var shift = e.shiftKey ? 1 : 0;
+		var numpad = (95 < e.keyCode && e.keyCode < 112) ? 1 : 0;
+		var escPressed = e.keyCode == 27 ? 1 : 0;
+		var sanitizedKey = e.key;
+		if (47 < e.keyCode && e.keyCode < 58) {
+			sanitizedKey = String.fromCharCode(e.keyCode);
+		}
+		else if (64 < e.keyCode && e.keyCode < 91) {
+			sanitizedKey = String.fromCharCode(e.keyCode);
+		}
+		var url = 'byond://?src=\ref[src];preference=keybindings_set;keybinding=[kb.name];old_key=[old_key];clear_key='+escPressed+';key='+sanitizedKey+';alt='+alt+';ctrl='+ctrl+';shift='+shift+';numpad='+numpad+';key_code='+e.keyCode;
+		window.location=url;
+		deedDone = true;
+	}
+	document.getElementById('focus').focus();
+	</script>
+	"}
+	winshow(user, "capturekeypress", TRUE)
+	var/datum/browser/popup = new(user, "capturekeypress", "<div align='center'>Keybindings</div>", 350, 300)
+	popup.set_content(HTML)
+	popup.open(FALSE)
+
+/datum/category_item/player_setup_item/controls/keybindings/OnTopic(href, list/href_list, mob/user)
+	switch(href_list["preference"])
+		if("keybindings_capture")
+			var/datum/keybinding/kb = global.keybindings_by_name[href_list["keybinding"]]
+			var/old_key = href_list["old_key"]
+			capture_keybinding(user, kb, old_key)
+			return TOPIC_REFRESH
+
+		if("keybindings_set")
+			var/kb_name = href_list["keybinding"]
+			if(!kb_name)
+				show_browser(user, null, "window=capturekeypress")
+				return TOPIC_REFRESH
+
+			var/clear_key = text2num(href_list["clear_key"])
+			var/old_key = href_list["old_key"]
+			if(clear_key)
+				if(pref.key_bindings[old_key])
+					pref.key_bindings[old_key] -= kb_name
+					if(!(kb_name in pref.key_bindings["None"]))
+						LAZYADD(pref.key_bindings["None"], kb_name)
+					if(!length(pref.key_bindings[old_key]))
+						pref.key_bindings -= old_key
+				show_browser(user, null, "window=capturekeypress")
+				user.client.set_macros()
+				SScharacter_setup.queue_preferences_save(pref)
```

Comment:
You probably shouldn't save here. They can save with the save button like for every other pref.